File: QE-Selenium-Python/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service
import time

driver = webdriver.Firefox()

try:
    driver.get("https://training-support.net/webelements/target-practice")

    print("Page Title:", driver.title)

    third_header = driver.find_element(By.XPATH, "(//h3)[1]")
    print("3rd Header Text:", third_header.text)

    fifth_header = driver.find_element(By.XPATH, "(//h3)[2]")
    print("5th Header Color:", fifth_header.value_of_css_property("color"))

    purple_button = driver.find_element(By.XPATH, "//button[text()='Purple']")
    print("Purple Button Classes:", purple_button.get_attribute("class"))

    slate_button = driver.find_element(By.XPATH, "//button[text()='Slate']")
    print("Slate Button Text:", slate_button.text)

    time.sleep(2)

finally:
    driver.quit()

File: QE-Selenium-Java/Activity4.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/target-practice");
            System.out.println("Page Title: " + driver.getTitle());

            // Correct header indexes
            WebElement firstHeader = driver.findElement(By.xpath("(//h3)[1]"));
            System.out.println("1st Header Text: " + firstHeader.getText());

            WebElement secondHeader = driver.findElement(By.xpath("(//h3)[2]"));
            System.out.println("2nd Header Color: " +
                    secondHeader.getCssValue("color"));

            WebElement purpleButton = driver.findElement(By.xpath("//button[contains(@class,'bg-purple')]"));
            System.out.println("Purple Button Classes: " +
                    purpleButton.getAttribute("class"));

            WebElement slateButton = driver.findElement(By.xpath("//button[contains(@class,'bg-slate')]"));
            System.out.println("Slate Button Text: " +
                    slateButton.getText());

        } finally {
            driver.quit(); // ALWAYS closes browser
        }
    }
}
